modules/monitoring-example-remote-write-authentication-settings.adoc

```diff
@@ -0,0 +1,340 @@
+kind: Secret
+metadata:
+  name: sigv4-credentials
+  namespace: openshift-monitoring
+type: Opaque
+----
+<1> Substitute your AWS API access key.
+<2> Substitute your AWS API secret key.
+
+.YAML for `sigv4` remote write authentication
+
+The following shows sample AWS Signature Version 4 authentication settings that use a `Secret` object named `sigv4-credentials` in the `openshift-monitoring` namespace:
+
+[source,yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cluster-monitoring-config
+  namespace: openshift-monitoring
+data:
+  config.yaml: |
+    prometheusK8s:
+      remoteWrite:
+      - url: "https://authorization.example.com/api/write"
+        sigv4:
+          region: <AWS_region> <1>
+          accessKey:
+            name: sigv4-credentials <2>
+            key: access
+          secretKey:
+            name: sigv4-credentials <2>
+            key: secret
+          profile: <AWS_profile_name> <3>
+          roleArn: <AWS_role_arn> <4>
+----
+<1> Substitute with your AWS region.
+If you do not enter a value, the region from the default credentials chain is used.
+<2> The name of the `Secret` object containing your AWS API access credential.
+<3> The name of the AWS profile being used to authenticate.
+<4> The unique identifier for Amazon Resource Name (ARN) assigned to your role.
+
+
+== Sample for basic authentication
+
+Basic authentication (`basicAuth`) sets the authorization header on every remote write request with the configured username and password.
+You can use this method together with bearer token authentication.
+
+.YAML for `Secret` object
+
+The following shows sample basic authentication settings for a `Secret` object named `rw-basic-auth` in the `openshift-monitoring` namespace: 
+
+[source,yaml]
+----
+apiVersion: v1
+stringData:
+  user: <basic_username> <1>
+  password: <basic_password> <2>
+kind: Secret
+metadata:
+  name: rw-basic-auth
+  namespace: openshift-monitoring
+type: Opaque
+----
+<1> Substitute your username.
+<2> Substitute your password.
+
+.YAML for `basicAuth` remote write authentication
+
+The following sample shows a `basicAuth` configuration that uses a `Secret` object named `rw-basic-auth` in the `openshift-monitoring` namespace.
+It assumes that you have already set up authentication credentials for the endpoint. 
+
+[source,yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cluster-monitoring-config
+  namespace: openshift-monitoring
+data:
+  config.yaml: |
+    prometheusK8s:
+      remoteWrite:
+      - url: "https://basicauth.example.com/api/write"
+        basicAuth:
+          username:
+            name: rw-basic-auth
+            key: user
+          password:
+            name: rw-basic-auth
+            key: password
+----
+
+== Sample for bearer token authentication
+
+Bearer token authentication sets the authorization header on every remote write request using the configured token.
+This method can be used together with basic authentication.
```

Comment:
Although this might be true, I don't see the point of adding the basicAuth configuration to the bearer token example. Do we have a reason to do it this way?

Reply:
deleted the basicAuth section from this sample